### PR TITLE
Renamed and updated

### LINKED ISSUE #12: rename to fastify-bankai
cc @yoshuawuyts

### PR DESCRIPTION
**Pull request by mcollina**
This would need another round of updates when bankai 9 comes out.

Fixes #12

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,24 @@
...
 ```
 
 ## Usage
 Simply require this plugin, pass the entry file and you are done!
 ```js
 const fastify = require('fastify')()
 
-fastify.register(require('fastify-pigeon'), {
+fastify.register(require('fastify-bankai'), {
   entryFile: './client.js'
```
Comment (delvedor) [inline README.md]: I think this should be just `entry` :)

File: index.js [hunk 2]
```diff
@@ -7,36 +7,44 @@ const fp = require('fastify-plugin')
...
   const html = !!opts.html
   const htmlPath = resolve(opts.html || '')
-  const assets = bankai(resolve(opts.entryFile || ''), opts.options || {})
+  const assets = bankai(resolve(opts.entry || ''), opts.options)
 
   fastify.get(opts.baseURL || '/', (req, reply) => {
     reply
       .header('Content-Type', 'text/html')
-      .send(html ? createReadStream(htmlPath) : assets.html())
+      .send(html ? createReadStream(htmlPath) : assets.html(req.req, req.res))
```
Comment (delvedor) [inline index.js]: We don't have `req.res`, probably you mean `reply.res`.

**Review by delvedor — COMMENTED** [review]
Since we are refactoring, probably we could add an `HEAD` route, what do you think?

**Comment by mcollina** [comment]
@delvedor definitely, but in another PR. Can you review it again? I've simplified things a bit.

File: index.js [hunk 3]
```diff
@@ -7,36 +7,40 @@ const fp = require('fastify-plugin')
...
   const htmlPath = resolve(opts.html || '')
-  const assets = bankai(resolve(opts.entryFile || ''), opts.options || {})
+  const assets = bankai(resolve(opts.entry || ''), opts)
 
-  fastify.get(opts.baseURL || '/', (req, reply) => {
+  fastify.get('/', (req, reply) => {
     reply
       .header('Content-Type', 'text/html')
-      .send(html ? createReadStream(htmlPath) : assets.html())
+      .send(html ? createReadStream(htmlPath) : assets.html(req.req, req.res))
```
Comment (delvedor) [inline index.js]: `reply.res` instead of `req.res`

File: index.js [hunk 4]
```diff
@@ -7,36 +7,40 @@ const fp = require('fastify-plugin')
...
+    return next(new Error('entry must be a string'))
+  }
+  if (opts.html && typeof opts.html !== 'string') {
+    return next(new Error('html must be a string'))
+  }
 
   const html = !!opts.html
   const htmlPath = resolve(opts.html || '')
-  const assets = bankai(resolve(opts.entryFile || ''), opts.options || {})
+  const assets = bankai(resolve(opts.entry || ''), opts)
```
Comment (delvedor) [inline index.js]: Do we need to remove the `prefix` key from the option object before passing it to `bankai`?

Comment (mcollina) [reply index.js]: done

**Review by delvedor — APPROVED** [review]
LGTM

**Pull request merged**